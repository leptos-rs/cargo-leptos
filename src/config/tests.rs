use super::Config;

fn opts(project: Option<&str>) -> crate::config::Opts {
    crate::config::Opts {
        release: false,
        js_minify: false,
        precompress: false,
        hot_reload: false,
        project: project.map(|s| s.to_string()),
        verbose: 0,
        features: Vec::new(),
        bin_features: Vec::new(),
        lib_features: Vec::new(),
        bin_cargo_args: None,
        lib_cargo_args: None,
        wasm_debug: false,
    }
}

// this test causes issues in CI because the tailwind tmp_file field is an absolute path,
// so differs by platform
/* #[test]
fn test_project() {
    let cli = opts(None);

    let conf = Config::test_load(cli, "examples", "examples/project/Cargo.toml", true);

    insta::assert_debug_snapshot!(conf);
} */

#[test]
fn test_workspace() {
    let cli = opts(None);

    let conf = Config::test_load(cli, "examples", "examples/workspace/Cargo.toml", true, None);

    insta::assert_debug_snapshot!(conf);
}

#[test]
fn test_workspace_project1() {
    let cli = opts(Some("project1"));

    let conf = Config::test_load(cli, "examples", "examples/workspace/Cargo.toml", true, None);

    insta::assert_debug_snapshot!(conf);
}

#[test]
fn test_workspace_project2() {
    let cli = opts(Some("project2"));

    let conf = Config::test_load(cli, "examples", "examples/workspace/Cargo.toml", true, None);

    insta::assert_debug_snapshot!(conf);
}

#[test]
fn test_workspace_in_subdir_project2() {
    let cli = opts(None);

    let conf = Config::test_load(
        cli,
        "examples/workspace/project2",
        "examples/workspace/Cargo.toml",
        true,
        None,
    );

    insta::assert_debug_snapshot!(conf);
}

#[test]
fn test_workspace_bin_args_project2() {
    let cli = opts(Some("project2"));

    let conf = Config::test_load(
        cli,
        "examples",
        "examples/workspace/Cargo.toml",
        true,
        Some(&["--".to_string(), "--foo".to_string()]),
    );

    insta::assert_debug_snapshot!(conf);
}
