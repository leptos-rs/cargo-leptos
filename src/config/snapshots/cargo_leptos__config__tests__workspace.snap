---
source: src/config/tests.rs
expression: conf
---
Config {
    projects: [
        Project {
            name: "project1",
            lib: LibPackage {
                name: "front-package",
                rel_dir: "project1/front",
                wasm_file: SourcedSiteFile {
                    source: "target/front/wasm32-unknown-unknown/debug/front_package.wasm",
                    dest: "target/site/project1/pkg/project1.wasm",
                    site: "pkg/project1.wasm",
                },
                js_file: SiteFile {
                    dest: "target/site/project1/pkg/project1.js",
                    site: "pkg/project1.js",
                },
                features: [],
                default_features: false,
                output_name: "project1",
                src_paths: "project1/app/src, project1/front/src",
                profile: Debug,
                ..
            },
            bin: BinPackage {
                name: "server-package",
                rel_dir: "project1/server",
                exe_file: "target/debug/server-package",
                target: "server-package",
                features: [],
                default_features: false,
                src_paths: "project1/app/src, project1/server/src",
                profile: Debug,
                bin_args: None,
                ..
            },
            style: StyleConfig {
                file: Some(
                    SourcedSiteFile {
                        source: "project1/css/main.scss",
                        dest: "target/site/project1/pkg/project1.css",
                        site: "pkg/project1.css",
                    },
                ),
                browserquery: "defaults",
                tailwind: None,
                site_file: SiteFile {
                    dest: "target/site/project1/pkg/project1.css",
                    site: "pkg/project1.css",
                },
            },
            watch: true,
            release: false,
            precompress: false,
            js_minify: false,
            hot_reload: false,
            site: Site {
                addr: 127.0.0.1:3000,
                reload: 127.0.0.1:3001,
                root_dir: "target/site/project1",
                pkg_dir: "pkg",
                file_reg: {},
                ext_file_reg: {},
            },
            end2end: None,
            assets: Some(
                AssetsConfig {
                    dir: "project1/assets",
                },
            ),
            ..
        },
        Project {
            name: "project2",
            lib: LibPackage {
                name: "project2",
                rel_dir: "project2",
                wasm_file: SourcedSiteFile {
                    source: "target/front/wasm32-unknown-unknown/debug/project2.wasm",
                    dest: "target/site/project2/pkg/project2.wasm",
                    site: "pkg/project2.wasm",
                },
                js_file: SiteFile {
                    dest: "target/site/project2/pkg/project2.js",
                    site: "pkg/project2.js",
                },
                features: [
                    "hydrate",
                ],
                default_features: false,
                output_name: "project2",
                src_paths: "project2/src",
                profile: Debug,
                ..
            },
            bin: BinPackage {
                name: "project2",
                rel_dir: "project2",
                exe_file: "target/debug/project2",
                target: "project2",
                features: [
                    "ssr",
                ],
                default_features: false,
                src_paths: "project2/src",
                profile: Debug,
                bin_args: None,
                ..
            },
            style: StyleConfig {
                file: Some(
                    SourcedSiteFile {
                        source: "project2/src/main.scss",
                        dest: "target/site/project2/pkg/project2.css",
                        site: "pkg/project2.css",
                    },
                ),
                browserquery: "defaults",
                tailwind: None,
                site_file: SiteFile {
                    dest: "target/site/project2/pkg/project2.css",
                    site: "pkg/project2.css",
                },
            },
            watch: true,
            release: false,
            precompress: false,
            js_minify: false,
            hot_reload: false,
            site: Site {
                addr: 127.0.0.1:3000,
                reload: 127.0.0.1:3001,
                root_dir: "target/site/project2",
                pkg_dir: "pkg",
                file_reg: {},
                ext_file_reg: {},
            },
            end2end: None,
            assets: Some(
                AssetsConfig {
                    dir: "project2/src/assets",
                },
            ),
            ..
        },
    ],
    cli: Opts {
        release: false,
        precompress: false,
        hot_reload: false,
        project: None,
        features: [],
        lib_features: [],
        lib_cargo_args: None,
        bin_features: [],
        bin_cargo_args: None,
        wasm_debug: false,
        verbose: 0,
        js_minify: false,
    },
    watch: true,
    ..
}
